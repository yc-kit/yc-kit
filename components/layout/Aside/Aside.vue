<template>
  <aside
    data-role="yc-layout-aside"
    class="w-64 flex items-center justify-center bg-blue-300"
  >
    Aside
  </aside>
</template>

<script lang="ts">
import { createComponent } from '@vue/composition-api'

// @vue/component
export default createComponent({
  name: 'YcLayoutAside'
})
</script>
