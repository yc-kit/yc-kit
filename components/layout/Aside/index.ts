export { default, default as YcLayoutAside } from './Aside.vue'
