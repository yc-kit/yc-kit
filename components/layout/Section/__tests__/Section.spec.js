import { shallowMount } from '@vue/test-utils'
import Setion from 'yc-layout/Section'
import Aside from 'yc-layout/Aside'

describe('Section', () => {
  const mountSection = () => {
    return shallowMount(Setion, {
      slots: {
        default: '<div>foo</div>'
      }
    })
  }

  test('render correctly', () => {
    const wrapper = mountSection()
    expect(wrapper.element).toMatchSnapshot()
  })

  test('should has correct value of "data-has-aside" property, when pass props "hasAside: true"', () => {
    const wrapper = mountSection()
    expect(wrapper.attributes('data-has-aside')).toBe(undefined)
    wrapper.setProps({ hasAside: true })
    wrapper.vm.$nextTick(() => {
      expect(wrapper.attributes('data-has-aside')).toBe('true')
    })
  })

  test('should has correct value of "data-has-aside" property, when slot include "Aside" component', () => {
    const wrapper = shallowMount(Setion, {
      slots: {
        default: Aside
      }
    })
    expect(wrapper.attributes('data-has-aside')).toBe(undefined)
    wrapper.vm.$nextTick(() => {
      expect(wrapper.attributes('data-has-aside')).toBe('true')
    })
  })
})
