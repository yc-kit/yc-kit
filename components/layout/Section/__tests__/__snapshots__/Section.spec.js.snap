// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section render correctly 1`] = `
<section
  class="flex flex-1 flex-col"
  data-role="yc-layout-section"
>
  <div>
    foo
  </div>
</section>
`;
